action.yml

```diff
@@ -20,7 +20,7 @@ inputs:
   git_ref:
     description: 'A TMT tests branch which will be used for tests'
     required: false
-    default: 'master'
+    default: '${{ github.base_ref }}'
```

Comment:
`base_ref` doesn't seem to be a good target for this:
> The base_ref or target branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is either pull_request or pull_request_target.

Meaning when run in `on.push` it would not be picked up correctly, and on `on.pull_request` it would not pick up the branch that is meant to be merged (instead it's the target branch)

Reply:
The problem is that there is no easy way to get the repo's default branch. I guess that this is the aim of your issue.

Reply:
When on PR, the `github.ref` expands to the branch from which the PR is opened. This is probably not what we want. But I'm not a user of this action. @phracek @zmiklank

Reply:
> for pull requests it is `refs/pull/<pr_number>/merge`

> When on PR, the `github.ref` expands to the branch from which the PR is opened.

From my reading of this it means that it is the branch on the target repo, containing the PR after merge. I.e. as if in this case you do:
```console
$ git clone https://github.com/sclorg/testing-farm-as-github-action
$ cd testing-farm-as-github-action
$ git fetch origin refs/pull/190/merge:pr_branch
$ git checkout pr_branch
$ git log
commit 8b4000a392891d7e4429697cdcb69086df958e8f (HEAD -> pr_branch)
Merge: 07b8a51 e75db73
Author: Jan Macku <jamacku@redhat.com>
Date:   Wed May 29 10:06:00 2024 +0000

    Merge e75db73f8bafbdd9511f4587431a2f67cf729177 into 07b8a51c96af484c108d3f41ff96780a7b2fe8b1

commit e75db73f8bafbdd9511f4587431a2f67cf729177
Author: Jan Macku <jamacku@redhat.com>
Date:   Wed May 29 10:25:40 2024 +0200

    feat: update default value of `git_ref`
    
    By default use `github.base_ref` and use `master` as a fallback.

commit 07b8a51c96af484c108d3f41ff96780a7b2fe8b1 (origin/main, origin/HEAD, main)
Author: Jan Macku <jamacku@redhat.com>
Date:   Wed May 29 09:17:42 2024 +0200

    feat: make `git_url` optional
    
    default value is `${{ github.server_url }}/${{ github.repository }}`
```

I believe this is always the default result we want.

---

The only uncertain case is `on.workflow_dispatch`, but for this case I think it is better to move the default from `action.yml` to the typescript accessing [`github.context.ref`](https://github.com/actions/toolkit/blob/main/packages/github/README.md) if `git_ref == ""` (the default if no `default` is specified). This will allow to write the following:
```
on:
  push: [ main ]
  pull_request:
  workflow_dispatch:
    inputs:
      git_ref:
        required: true

jobs:
  testing_farm:
    steps:
      - uses: sclorg/testing-farm-as-github-action@v2
        with:
          git_url: ${{ inputs.git_ref }}
```
(here `${{ inputs.git_ref }}` will expand to empty when the other triggers are called, and we use `github.context.ref` == `GITHUB_REF` == `${{ github.ref }}` for the default behavior)

Reply:
> When on PR, the `github.ref` expands to the branch from which the PR is opened. This is probably not what we want. But I'm not a user of this action. @phracek @zmiklank

Yeah, correct. we want to clone it in to branch from which specific PR is opened.

Reply:
In case you would like to both, like  `on.push`  and `on.pull_request` , I would recommend to split them into two GitHub Actions.

Reply:
> In case you would like to both, like `on.push` and `on.pull_request` , I would recommend to split them into two GitHub Actions.

I'm afraid that would make maintenance rather cumbersome, and it would not look good when compared with other github actions like `actions/checkout`, `ci-build-wheel`, `cirrus-ci`, etc.

A simple example is when someone is trying to work on the `.github/workflows` itself. We would like the PR work to be as equivalent as the workflow used for `on.push`.

Reply:
@LecrisUT You are right. `github.ref` might be a better value for `git_ref`.